### ISSUE TITLE
Unable to import frameworks into Swift project.

### ISSUE THREAD

**Issue by dtrenz** [spotify/ios-sdk#811]
## Description
I'm unable to import any of the Spotify frameworks into my Swift files. Are these frameworks set with "Defines Module" enabled (see: [Apple docs](https://developer.apple.com/library/content/documentation/Swift/Conceptual/BuildingCocoaApps/MixandMatch.html#//apple_ref/doc/uid/TP40014216-CH10-ID134))?

### Versions
beta 25

### Severity
Major

## Steps to Reproduce
1. Add all 3 Spotify frameworks to a Swift iOS project/target.
1. Add`-ObjC` to "Other Linker Flags" in the target's build settings.
1. Attempt to import any of the frameworks into a Swift file in the target (e.g. `import SpotifyAudioPlayback`)

### Actual behaviour
> No such module 'SpotifyAudioPlayback'

### Expected behaviour
Project builds, and classes/types from the frameworks should be accessible from within Swift files.

**Comment by davidseek** [comment]
I have set up a starter project for you, so you see what you have to do. It's pretty simple. Put the desired Frameworks into your Project. Create a Bridging Header, or use mine. 

Then you need to go to Build Settings and set the Bridging Header:

<img width="1152" alt="screen shot 2016-12-06 at 9 34 36 pm" src="https://cloud.githubusercontent.com/assets/17073950/20942484/d20c81d2-bbfb-11e6-8523-87e77bb59082.png">

Last step is to put the desired Frameworks into the Bridging Header file:

```
#import <SpotifyAuthentication/SpotifyAuthentication.h>
#import <SpotifyMetadata/SpotifyMetadata.h>
#import <SpotifyAudioPlayback/SpotifyAudioPlayback.h>
```

That's it. From now on, you can use the Frameworks **without import**, just use them.
You will see in the starter project.

Have fun.

**Comment by dtrenz** [comment]
Interesting. I tried doing exactly this before, and it did not work. I will try again, thank you.

However, it would be best if these frameworks could be configured to be importable into Swift projects.

**Comment by davidseek** [comment]
**That is** how they are importable into Swift. Other option would be to use them with Cocoapods. Both is working perfectly. Using them in several projects and I'm coding in Swift only.

**Comment by dtrenz** [comment]
Well, I think the problem is that I'm trying to use these frameworks inside of my own framework. I don't believe that frameworks cannot use Bridging Headers.

**Comment by davidseek** [comment]
That is called a dependency. And is no problem.
There are many threads on SO to that issue.

f.e.: http://stackoverflow.com/questions/28815487/building-a-swift-framework-with-references-to-objective-c-code

**Comment by dtrenz** [comment]
Right, that is what I am trying to do, but it doesn't work. I believe that this is because the frameworks are not built with "Defines Module" enabled (see OP).

**Comment by escamoteur** [comment]
@davidseek Does the SPTAuthViewController work without problems for you?

**Comment by davidseek** [comment]
yes.

**Comment by zgthompson** [comment]
@dtrenz I was able to resolve your issue by creating a modulemap, which I believe would have been auto-generated had the framework been built with "Defines Module" enabled. No problem though, this can be done manually. Using SpotifyAudioPlayback.framework as an example:

1. Inside SpotifyAudioPlayback.framework create a folder called Modules.

2. Inside Modules create a file called module.modulemap

3. Edit module.modulemap to contain the following text:

```
framework module SpotifyAudioPlayback {
    umbrella header "SpotifyAudioPlayback.h"

    export *
    module * { export * }
}
```

The framework should now be importable inside your own framework.

**Comment by paperview** [comment]
@zgthompson how do you add Modules directory to the framework?  Xcode will not let me drop in a folder there or add a directory next to Headers
